### ISSUE TITLE
NixOS manual: docbook schema configuration uses missing path

### ISSUE THREAD

**Issue by jmitchell** [NixOS/nixpkgs#24959]
## Issue description

While following the instructions in the manual to [support DocBook 5 XML editing in Emacs](https://nixos.org/nixos/manual/index.html#sec-emacs-docbook-xml) I discovered the recommended schema configuration doesn't work when `docbook5` is installed via `environment.systemPackages`. Specifically, `/run/current-system/sw/share/xml/docbook-5.0/rng/docbookxi.rnc` doesn't exist.

### Steps to reproduce

1. add `docbook5` to `environment.systemPackages` in `/etc/nixos/configuration.nix`
2. run `nixos-rebuild switch`
3. add `(add-to-list 'rng-schema-locating-files "~/.emacs.d/schemas.xml")` to Emacs configuration
4. write the following to `~/.emacs.d/schemas.xml`

```
<?xml version="1.0"?>
<locatingRules xmlns="http://thaiopensource.com/ns/locating-rules/1.0">
  <namespace ns="http://docbook.org/ns/docbook"
             uri="/run/current-system/sw/share/xml/docbook-5.0/rng/docbookxi.rnc"/>
</locatingRules>
```

5. open `$NIXPKGS/doc/manual.xml` in Emacs.
6. go to `*Messages*` buffer and see the following error:

```
File mode specification error: (file-error Opening input file No such file or directory /run/current-system/sw/share/xml/docbook-5.0/rng/docbookxi.rnc)
```

7. notice that `/run/current-system/sw/share` has no `xml` directory.

## Workaround

Use the other suggested approach where `docbook5` is installed ad-hoc and the `docbookxi.rnc` path in the `schemas.xml` points to user profile instead.

## Technical details

* System: 17.03.945.5acb454e2a (Gorilla)
* Nix version: nix-env (Nix) 1.11.8
* Nixpkgs version: "17.03.945.5acb454e2a"

**Comment by rvl** [comment]
I have the file `/run/current-system/sw/share/xml/docbook-5.0/rng/docbookxi.rnc`.

When I remove `pkgs.docbook5` from `environment.systemPackages` and `nixos-rebuild switch`, the directory `/run/current-system/sw/share/xml/docbook-5.0` disappears.

Is docbook5 in the list when you run `nixos-option environment.systemPackages | grep --color docbook5`?

**Comment by jmitchell** [comment]
> Is docbook5 in the list when you run `nixos-option environment.systemPackages | grep --color docbook5`?

Yes, but even so the `/run/current-system/sw/share` directory still has no `xml` directory. Steps to ensure my environment is properly cleaned and set up to have `docbook5` in the configured system packages.

1. `nix-env -e docbook5`
2. removed `pkgs.docbook5` from `environment.systemPackages`
3. `nix-collect-garbage`
4. confirmed your suggested command returns nothing
5. added `pkgs.docbook5` back to configured `environment.systemPackages` and run `nixos-rebuild switch`
6. ran your command again and saw `"/nix/store/vv6pg515z9n70drggdpc9h04aw6zmlv5-docbook5-5.0"` in the output
7. confirmed `"/nix/store/vv6pg515z9n70drggdpc9h04aw6zmlv5-docbook5-5.0` exists, but `/run/current-system/sw/share/xml` doesn't.

**Comment by rvl** [comment]
Hi @jmitchell, Just checking, do you now have `/run/current-system/sw/share/xml/docbook-5.0` if `pkgs.docbook5` is in `environment.systemPackages`?

**Comment by jtojnar** [comment]
I would suspect that this only works for some users because some module puts `/share` into their `environment.pathsToLink`. I believe this kind of carpet bombing is not correct, see https://github.com/NixOS/nixpkgs/issues/47173.

Instead of setting global paths in you profile, I would recommend to follow the suggestion in https://github.com/NixOS/nix-pills#development, along with a simplified config from https://github.com/NixOS/nixpkgs/issues/68236
